### DISCUSSION TITLE
Move due date to one day after start, when start is moved after due

### DISCUSSION THREAD

**Discussion by patrickunterwegs** [TechbeeAT/jtxBoard#19] [Polls]
In #15  the following idea came up (thx @xcy7e): 

> > To this topic I had the suggestion, of giving the user more freedom when setting dates, because moving the start date on a later date always requires to set the end date after the desired start date beforehand, which is uncommon usage behaviour. I'd expect to have the end date automatically moved 1 day behind the start date if it's being moved.

You mean the start and due date here, right? (because the completed date can anyway be set independently). 

The basic problem here is that the standard defines, that the due date MUST be after the start date. If the due and start date(-time) are equal or if the due date is before the start date, servers would usually throw an error. So this should absolutely be avoided.

When thinking of automatically updating the due date to one day after the start date, it seems strange to me to choose one day as default - why not one hour, why not one week, ...? So setting any date just seems to end up pretty random for the user. I am also questioning how ofter you would actually run into that case so that it would get bothering to update first the due date and then the start date.
But I am open for discussion, so I would like to start this poll.

**Comment by xcy7e** [comment]
To my understanding, user *must* set the 2nd dropdown (due date) *before* the 1st dropdown (start date) - if start has to be later than current due, right? This feels a little like a break in flow to me. If you know it, its fine, but I think the average user would run into this at least once 

You might **not** even have to choose a **fixed** a **timespan** to shift the due date. Just shift it by the same time distance it already has.
e.g. (start: 01.01.23, due: 03.01.23) manually shifted start to (start: 04.01.23) (+3d) then automatically shift due to (due: 06.01.23) (+3d)
and for time the same logic - keep the distance and align the shift.

Just an Idea.

**Reply by steinhmor** [reply]
This is a behaviour I notice in many calendar apps and would on the one hand be familiar and on the other hand sometimes drives me crazy (when the due date always is changing when the start date is changed - this should be avoided in my opinion). Maybe an explanation for the partly blocked date-selection-dialog for the start date would already reduce the confusion?
